src/nft/types/collection/collection.ts

```diff
@@ -83,31 +85,29 @@ export interface TokenRarity {
 }
 
 export interface TokenMetadata {
-  name: string
-  imageUrl: string
-  smallImageUrl: string
-  metadataUrl: string
-  rarity: TokenRarity
-  suspiciousFlag: boolean
-  suspiciousFlaggedBy: string
-  standard: TokenType
+  name?: string
+  imageUrl?: string
+  smallImageUrl?: string
+  metadataUrl?: string
+  rarity?: TokenRarity | Rarity
+  suspiciousFlag?: boolean
+  standard?: TokenType | NftStandard
 }
 
+// TODO when deprecating activity query, remove all outdated types (former in optional fields)
 export interface ActivityEvent {
-  collectionAddress: string
+  collectionAddress?: string
   tokenId?: string
   tokenMetadata?: TokenMetadata
-  eventType: ActivityEventType
-  marketplace?: Markets
-  fromAddress: string
+  eventType?: NftActivityType
+  marketplace?: Markets | string
+  fromAddress?: string
   toAddress?: string
   transactionHash?: string
-  orderHash?: string
-  orderStatus?: OrderStatus
-  price?: string
+  orderStatus?: QueryOrderStatus | OrderStatus
```

Comment:
will QueryOrderStatus not map directly to OrderStatus this whole double checking might not be necessary

Reply:
No it doesn't map directly